Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1358,6 +1358,18 @@ void WebProcessProxy::postMessageToRemote(WebCore::ProcessIdentifier destination
         webProcessProxy->send(Messages::WebProcess::RemotePostMessage(identifier, target, message), 0);
 }
 
+void WebProcessProxy::renderTreeDump(WebCore::ProcessIdentifier destinationProcessIdentifier, WebCore::FrameIdentifier frameIdentifier, int baseIndent, CompletionHandler<void(String)>&& completionHandler)
+{
+    String result;
+    auto webProcessProxy = processForIdentifier(destinationProcessIdentifier);
+    if (webProcessProxy) {
+        auto reply = webProcessProxy->sendSync(Messages::WebProcess::RenderTreeDump(frameIdentifier, baseIndent), 0);
+        if (reply)
+            std::tie(result) = reply.takeReply();
```

Comment:
This use of std::tie is a little strange.  Can we just return completionHandler(reply.takeReply()) instead?

Also, early return completionHandler({ }) if webProcessProxy or reply aren't truthy to reduce indentation and keep code flow more straightforward.

Reply:
`reply.takeReply()` returns a tuple, which is why I needed this. I'm seeing the same pattern at other places in case of sync messages, am I missing something?